### PR TITLE
feat(component): new week-date-picker component

### PR DESCRIPTION
**Pull request by CatsJuice**
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://github.com/toeverything/AFFiNE/assets/39363750/49d7a1ee-2832-4b61-a427-e627dae92952">
  
</picture>

### CODE REVIEW

**Comment by CatsJuice** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5468** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5477** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5477?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5517** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5517?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5549** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #5557** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #5558** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5558?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #5559** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5559?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #5574** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5574?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #5578** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5578?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #5579** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5579?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5477?utm_source=stack-comment).

File: packages/frontend/component/src/ui/date-picker/week-date-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  // when value changes, reset cursor
+  useEffect(() => {
+    value && setCursor(dayjs(value));
+  }, [value]);
+
+  // TODO: keyboard navigation
+  useEffect(() => {
+    if (!weekRef.current) return;
+    const el = weekRef.current;
+    const onKeyDown = (e: KeyboardEvent) => {
```
Comment (pengx17) [inline packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:149]: may need to stop event propagation if it is handled

**Comment by joooye34** [comment]
Seems to have some UX problems, but fix later is fine.

**Review by joooye34 — APPROVED** [review]

File: packages/frontend/component/src/ui/date-picker/week-date-picker.tsx [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+      updateRange([0, viewPortSize]);
+    } else {
+      const end = Math.min(range[1] + viewPortSize, 7);
+      const start = Math.min(range[0] + viewPortSize, end - viewPortSize);
+      updateRange([start, end]);
+    }
+  }, [cursor, range, updateRange, viewPortSize]);
+  const onPrev = useCallback(() => {
+    if (viewPortSize === 7) {
+      setCursor(cursor.add(-1, 'week'));
```
Comment (joooye34) [inline packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:78]: When I use the keyboard to go to the previous or next week, the focus will be lost and can't action continuously

Comment (CatsJuice) [reply packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:78]: Yes, the implementation of keyboard navigation requires a better solution, so I marked it as `// TODO`.

File: packages/frontend/component/src/ui/date-picker/week-date-picker.tsx (hunk 1 above)
Comment (joooye34) [reply packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:149]: is it ok？not sure

Comment (joooye34) [reply packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:149]: cc @CatsJuice

Comment (CatsJuice) [reply packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:149]: It seems fine at the moment, but adding it might be safer.

Comment (CatsJuice) [reply packages/frontend/component/src/ui/date-picker/week-date-picker.tsx:149]: Added

**Pull request merged**